scripts/run-executable.ps1

```diff
@@ -378,10 +415,30 @@ function Wait-Executable($Exe) {
             }
         }
         $Exe.Process.WaitForExit()
-        if ($Exe.Process.ExitCode -ne 0) {
-            LogErr "Process had nonzero exit code: $($Exe.Process.ExitCode)"
-            $KeepOutput = $true
+        $exitCode = $Exe.Process.ExitCode
+
+        # When using ProcDump, the process is ProcDump itself, not the target.
+        # The ProcDump exit codes does not map cleanly to the status of the target process, so we will just log it.
+        if (!$UseProcDump -or !$IsWindows) {
+            # Normal process (not wrapped by ProcDump)
+            if ($exitCode -ne 0) {
+                LogErr "Process had nonzero exit code: $exitCode"
+                $KeepOutput = $true
+            }
+        } else {
+            Log "ProcDump exit code: $exitCode"
```

Comment:
If the process return an error code but doesn't crash, does this mean we don't set `KeepOutput` to true, and won't be keeping logs?

If we can't know the target return code when called with procdump, maybe we should use it as a monitor instead (not `-x` + run the target separately)

Reply:
> > If the process returns an error code, ProcDump will generate a dump
> 
> Are you sure about that? If the program returns (not crash), what is there to dump?
> 
> > I am not sure if doing 2 concurrent runs in the same script is the way to go.
> 
> That's not what I am suggesting. `ProcDump` without the `-x` option runs in a "monitor" mode. It will wait for a process of the expected name to start, attach to it, and create dumps if it fails with what procdump is configured to monitor.

Launching procdump as a sidecar was how I tried implementing it originally, but it was running into too many race conditions, making it unreliable. If the main issue is we need to extract what the inner process's exit code is, we can just parse it from stdout. Procdump prints that there.